tools/c7n_azure/c7n_azure/container_host/policy.py

```diff
@@ -0,0 +1,124 @@
+from c7n.policy import PullMode, ServerlessExecutionMode, execution
+
+
+class AzureContainerHostMode(ServerlessExecutionMode):
+    """A policy that runs/executes in container-host mode."""
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'execution-options': {'type': 'object'}
+        }
+    }
+
+    POLICY_METRICS = ('ResourceCount', 'ResourceTime', 'ActionTime')
+
+    def __init__(self, policy):
+        self.policy = policy
+        self.log = logging.getLogger('custodian.azure.AzureContainerHostMode')
+
+    def run(self, event=None, lambda_context=None):
+        raise NotImplementedError("subclass responsibility")
+
+    def provision(self):
+        pass
+
+
+@execution.register(CONTAINER_TIME_TRIGGER_MODE)
+class AzureContainerPeriodicMode(AzureContainerHostMode, PullMode):
+    """A policy that runs at specified time intervals."""
+    schema = utils.type_schema(CONTAINER_TIME_TRIGGER_MODE,
+                               schedule={'type': 'string'},
+                               rinherit=AzureContainerHostMode.schema)
+
+    def provision(self):
+        super(AzureContainerPeriodicMode, self).provision()
+
+    def run(self, event=None, lambda_context=None):
+        """Run the actual policy."""
+        return PullMode.run(self)
+
+    def get_logs(self, start, end):
+        """Retrieve logs for the policy"""
+        raise NotImplementedError("error - not implemented")
+
+
+@execution.register(CONTAINER_EVENT_TRIGGER_MODE)
+class AzureContainerEventMode(AzureContainerHostMode):
+    """A policy that runs at specified time intervals."""
+    schema = utils.type_schema(CONTAINER_EVENT_TRIGGER_MODE,
+                               events={'type': 'array', 'items': {
+                                   'oneOf': [
+                                       {'type': 'string'},
+                                       {'type': 'object',
+                                        'required': ['resourceProvider', 'event'],
+                                        'properties': {
+                                            'resourceProvider': {'type': 'string'},
+                                            'event': {'type': 'string'}}}]
+                               }},
+                               rinherit=AzureContainerHostMode.schema)
+
+    def provision(self):
+        super(AzureContainerEventMode, self).provision()
+
+    def run(self, event=None, lambda_context=None):
+        resources = self.policy.resource_manager.get_resources([event['subject']])
+
+        resources = self.policy.resource_manager.filter_resources(
+            resources, event)
+
+        if not resources:
+            self.policy.log.info(
+                "policy: %s resources: %s no resources found" % (
+                    self.policy.name, self.policy.resource_type))
+            return
+
+        resources = self.policy.resource_manager.filter_resources(
```

Comment:
This is all just straight copy from the azure functions code.... :(

Reply:
I was trying to avoid taking on refactoring all those policy files correctly but I guess since the code is wrong everywhere I should try to go ahead and refactor into some common base and fix/test it all.  Ugh.

Reply:
lets fix it with a separate pr.

Reply:
I spent a while trying to find a good way to solve this with multiple inheritance, but we have such a mess of inheritance already with the way the functions ones are it would be nearly a whole rewrite to make it look good, so instead I just made a common utility class and moved the event run there, so this is fixed everywhere now.